service/middleware.go

```diff
@@ -250,9 +250,9 @@ func createProxyRequestMiddleware(next http.Handler, config config.Config, servi
 				for headerName, headerValue := range typedCachedResponse.HeaderMap {
 					w.Header().Add(headerName, headerValue)
 				}
-				// add CORS headers
+				// add CORS headers (if not already added)
 				accessControlAllowOriginValue := config.GetAccessControlAllowOriginValue(r.Host)
-				if accessControlAllowOriginValue != "" {
+				if w.Header().Get("Access-Control-Allow-Origin") == "" && accessControlAllowOriginValue != "" {
```

Comment:
add some comments so future devs will take less time to understand this line then it took us to fix please

Reply:
also good to update docs for cache for how it interacts with cors headers

Reply:
yeah, I'll do just want to make sure, it's correct fix

Reply:
@galxy25  https://github.com/Kava-Labs/kava-proxy-service/pull/75